### PR TITLE
[WIP] Ansible playbook for YasOpenstack

### PR DESCRIPTION
**Pull request by carawarner**
## SUMMARY

**Fixes**
https://refinery29.atlassian.net/browse/INF-1300

**Releases** 
https://refinery29.atlassian.net/browse/INF-1234
https://refinery29.atlassian.net/browse/INF-1406

## TODO

- [x] Using `ansible-galaxy` generate scaffolding for new role `yas-openstack`.
- [x] Using `ansible-galaxy` import roles for `yas` and `pyenv` from the yas repository.
- [x] Start fleshing out `yas-openstack` role: make `yas` a dependency, clone the YasOpenstack repository, and install YasOpenstack.

Figure out...

- [ ] Are the Slackbot credentials being set? The `yas` role does some of that.
- [ ] Does it matter where the `yas` and `YasOpenstack` repositories are cloned to? Right now `yas` is ending up in `/tmp/yas` and `YasOpenstack` is in `/srv/YasOpenStack`. 

## Testing

TODO

### CODE REVIEW

**Comment by carawarner** [comment]
I've got a playbook that clones and installs YasOpenstack and yas: 
ansible-playbook -i yasop.cloud.rf29.net, playbooks/main.yml

@schlueter I have specific questions: 
1. Does it matter what directory I pick as the target when cloning yas and YasOpenstack? 
2. I'm getting `systemctl: command not found` when I try to `restart yas` on the OpenStack instance where I am currently testing: yasop.cloud.rf29.net. Why is it missing?

File: playbooks/main.retry [hunk 1]
```diff
@@ -0,0 +1 @@
+yasop.cloud.rf29.net
```
Comment (schlueter) [inline playbooks/main.retry]: Don't check in retry files. You can put it in a global gitignore: ```$ git config -l | grep gitignore
core.excludesfile=~/.gitignore_global
~/w/r/vagrant.local.rf29.net  master
$ grep retry ~/.gitignore_global
*.retry
```

File: playbooks/main.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+- hosts: all
+  gather_facts: no
+  become: yes
+  tasks:
+    - name: apt-get update
+      raw: apt-get update -qq
+    - name: Install python 2.7
+      raw: apt-get install -qq python-minimal
```
Comment (schlueter) [inline playbooks/main.yml]: This is only required on xenial, and should be in a separate playbook.

File: playbooks/roles/yas-openstack/meta/main.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+dependencies:
+  - name: yas
+    src: https://github.com/refinery29/yas.git
+    scm: git
+    dep_msg: "Hello from test-role"
```
Comment (schlueter) [inline playbooks/roles/yas-openstack/meta/main.yml]: wip?

File: playbooks/roles/yas/meta/.galaxy_install_info [hunk 4]
```diff
@@ -0,0 +1 @@
+{install_date: 'Wed Aug  2 20:12:09 2017', version: ''}
```
Comment (schlueter) [inline playbooks/roles/yas/meta/.galaxy_install_info]: I don't think we need this. Where'd it come from?

File: playbooks/roles/yas/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    accept_hostkey=yes
+
+- name: Install yas
+  pip:
+    name={{ local_repository }}
+    state=latest
+    executable={{ pip_executable }}
+
+- name: Set slack_app_token if provided
+  lineinfile:
```
Comment (schlueter) [inline playbooks/roles/yas/tasks/main.yml]: We should use a template for this file.

**Comment by schlueter** [comment]
Ah, the ansible-galaxy commands. We typically just create the directory structure manually. 

Addressing the questions:

1. We don't need to download, we can just use something from this stuff I did
```
ubuntu@foo:~$ history (minus some stuff)
    5  sudo add-apt-repository ppa:fkrull/deadsnakes
    6  sudo apt update
    8  sudo apt search python3
    9  sudo apt install python3.6
   16  python3.6 -m pip install -U pip setuptools
   17  curl -O https://bootstrap.pypa.io/get-pip.py
   18  less get-pip.py
   19  python3.6 get-pip.py
   20  sudo python3.6 get-pip.py
   24  pip3 install git+https://github.com/refinery29/yas.git
   25  sudo pip3 install git+https://github.com/refinery29/yas.git
   28  sudo apt install libsystemd-dev
   29  sudo pip3 install git+https://github.com/refinery29/yas.git
   30  sudo apt install libpython3.6-dev
   31  sudo pip3 install git+https://github.com/refinery29/yas.git
```
2. Only an issue on trusty

File: playbooks/roles/yas/tasks/main.yml (hunk 5 above)
Comment (carawarner) [inline playbooks/roles/yas/tasks/main.yml]: `yas/tasks/main.yml` was imported along with the rest of the `yas` role using ansible-galaxy. Does that change anything?

File: playbooks/main.yml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+---
+- hosts: all
+  gather_facts: no
+  become: yes
+  tasks:
+    - name: apt-get update
+      raw: apt-get update -qq
```
Comment (schlueter) [inline playbooks/main.yml]: This is usually unnecessary as well, as specifying `update_cache: yes` in an Ansible apt module call will do it before anything that needs it.

**Pull request closed without merging**